task/github-request-reviewers/0.1/github-request-reviewers.yaml

```diff
@@ -0,0 +1,110 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-request-reviewers
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: github
+    tekton.dev/displayName: "request github reviewers"
+spec:
+  description: >-
+    This Task will request reviewers for a GitHub PR
+
+    It can take either a list of users or teams or both.
+    Note that, when teams are used, GitHub does not verify
+    if the team (or any of its members) are collaborators
+    the repo, so a 201 reply may still result in no reviewer
+    added to the PR.
+  workspaces:
+    - name: github
+      description: The workspace that contains the GitHub secret
+  params:
+    - name: GITHUB_HOST_URL
+      description: |
+        The GitHub host, adjust this if you run a GitHub Enteprise.
+      default: "api.github.com"
+      type: string
+
+    - name: API_PATH_PREFIX
+      description: |
+        The API path prefix, GitHub Enterprise has a prefix e.g. /api/v3
+      default: ""
+      type: string
+
+    - name: PACKAGE
+      description: "{owner}/{repo}"
+      type: string
+
+    - name: PULL_REQUEST_NUMBER
+      description: The pull request number
+      type: string
+
+    - name: REVIEWERS
+      description: |
+        Command separated list of github user slugs.
+      type: string
+      default: ""
+
+    - name: TEAM_REVIEWERS
+      description: |
+        Command separated list of github team slugs.
+      type: string
+      default: ""
+
+    - name: GITHUB_TOKEN_FILE
+      description: |
+        The file within workspace that contains the GitHub token.
+      type: string
+      default: token
+
+  steps:
+  - name: set-reviewers
+    image: python:3.9-alpine3.13
+    script: |
+      #!/usr/bin/env python
+      import json
+      import http.client
+
+      # Load the GitHub Token
+      github_token = ""
+      with open("$(workspaces.github.path)/$(params.GITHUB_TOKEN_FILE)", "r") as gh:
+        github_token = gh.read()
```

Comment:
any reason for reading the secrets via this reading the file and not mount the secrets to env and then taking the value from there?

Reply:
I think having a workspace is more flexible - you can still bind it to a k8s secret but you could also have something else writing your secret to the volume (e.g. from some kind of vault).